### PR TITLE
Backport `pallet-identity` double encoding on signature payload fix into crates io release 1.7.0

### PR DESCRIPTION
**Pull request by georgepisaltu**
Backport for #4646

### CODE REVIEW

**Comment by Morganamilo** [comment]
This change is marked as major so is not suitable for backport. The existing release branches should stay stable and have no breaking changes.

**Comment by georgepisaltu** [comment]
Ok, closing the PR then

**Comment by joepetrowski** [comment]
@Morganamilo we have had some other "major" backports, even that changed the `Config` trait types. I know it is not perfect but this is important to ensure that actual breaking changes don't happen on the People Chain, and it's a good opportunity to do it before the affected feature is even in production at all.

File: substrate/frame/identity/Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "pallet-identity"
-version = "29.0.1"
+version = "30.0.0"
```
Comment (acatangiu) [inline substrate/frame/identity/Cargo.toml:3]: `30.0.0` is already released (guessing with sdk v1.8.0)

```suggestion
version = "29.0.2"
```

**Review by acatangiu — APPROVED** [review]
In this case we can keep everything sane by yanking the previous version of `pallet-identity` and releasing this new "patched" one.

The `major` breaking change here will not break compilation (API is unchanged); the interpretation of one of the parameters does indeed change, so it is only breaking for actual users of it - which there are none yet since this hasn't been deployed anywhere yet.

Patching the fix here allows us to use the right behavior on the first deployment.

File: substrate/frame/identity/Cargo.toml (hunk 1 above)
Comment (georgepisaltu) [reply substrate/frame/identity/Cargo.toml]: Looks like latest on crates.io is v34.0.0, so there's some versions in between. Anyway, updated it to v29.0.2.

**Comment by Morganamilo** [comment]
> In this case we can keep everything sane by yanking the previous version of `pallet-identity` and releasing this new "patched" one.
> 
> The `major` breaking change here will not break compilation (API is unchanged); the interpretation of one of the parameters does indeed change, so it is only breaking for actual users of it - which there are none yet since this hasn't been deployed anywhere yet.
> 
> Patching the fix here allows us to use the right behavior on the first deployment.

If this crate is considered unreleased then in this case it's fine. But I don't know what you mean by it hasn't been deployed yet.

**Comment by georgepisaltu** [comment]
> it is only breaking for actual users of it - which there are none yet since this hasn't been deployed anywhere yet

Technically it is deployed on Kusama People chain, but there is no way to use the broken functionality because there are no registered username authorities. It is unreleased on Polkadot People chain, which is good because we have a chance to fix it. I think yanking the current version makes sense and we can just release this "patch" version on top of the yanked one, and do the same for the remaining release branches.

**Review by seadanda — APPROVED** [review]

**Pull request closed without merging**